### PR TITLE
avoid non necessary objects allocation if session do not have any attributes data

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: olivier lamy <oliver.lamy@gmail.com>

### CODE REVIEW

File: tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/JdbcTestHelper.java [hunk 1]
```diff
@@ -173,17 +173,17 @@ public static boolean checkSessionPersisted(SessionData data)
...
-                " where " + ID_COL + " = ? and " + CONTEXT_COL +
-                " = ? and virtualHost = ?");
-            statement.setString(1, data.getId());
-            statement.setString(2, data.getContextPath());
-            statement.setString(3, data.getVhost());
+            statement = con.prepareStatement(
+                "select * from " + TABLE +
+                    " where " + ID_COL + " = ? and " + CONTEXT_COL +
+                    " = ? and virtualHost = ?" );
+            statement.setString(1, data.getId() );
```
Comment (janbartel) [inline tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/JdbcTestHelper.java]: how are you getting away with the extra space before the final ); ??? Is checkstyle broken?

File: jetty-server/src/main/java/org/eclipse/jetty/server/session/JDBCSessionDataStore.java [hunk 2]
```diff
@@ -737,20 +745,26 @@ protected void doUpdate(String id, SessionData data)
...
+                        byte[] bytes = baos.toByteArray();
+                        try (ByteArrayInputStream bais = new ByteArrayInputStream(bytes))
+                        {
+                            statement.setBinaryStream( 7, bais, bytes.length );//attribute map as blob
+                        }
                     }
                 }
+                else
+                {
+                    statement.setBinaryStream( 7, EMPTY, 0);
```
Comment (janbartel) [inline jetty-server/src/main/java/org/eclipse/jetty/server/session/JDBCSessionDataStore.java:763]: Does this deserialize correctly? Serializing an empty map would be a different result than an empty byte string.

**Review by janbartel — APPROVED** [review]
Fine, subject to ensuring that serializing 0 bytes instead of an empty map deserializes ok.

File: tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/JdbcTestHelper.java (hunk 1 above)
Comment (olamy) [reply tests/test-sessions/test-jdbc-sessions/src/test/java/org/eclipse/jetty/server/session/JdbcTestHelper.java]: ohh my bad I usually avoid only blank line. But agree I have to take care of spaces as well

File: jetty-server/src/main/java/org/eclipse/jetty/server/session/JDBCSessionDataStore.java (hunk 2 above)
Comment (olamy) [reply jetty-server/src/main/java/org/eclipse/jetty/server/session/JDBCSessionDataStore.java:763]: just an empty map but good point I will add a test for that.

**Pull request merged**